gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -191,11 +181,6 @@ protected void processMessage(DecodeableKafkaRecord<byte[],byte[]> message) {
       }
     }
 
-    if (gobblinTrackingEvent.getName().equals(JobEvent.WORK_UNITS_CREATED)) {
-      emitWorkUnitCountMetric(gobblinTrackingEvent);
-      return;
-    }
```

Comment:
please explain what semantics we're changing by removing this... and why

e.g. were we previously 'mirroring'/'repeating' the WU event publication on gaas, in addition to what the executor already published?

Reply:
Before the executor did not publish this metric, and I'm proposing that it does instead of the JobMonitor.
Since the job monitor runs on every GaaS instance, for a single pipeline the metrics would be emitted for each of the GaaS hosts multiple times. By moving it to the executor, it only gets emitted once by a single host which makes the metric much more cleaner to track.